docs/PacketFence_MSPKI_Quick_Install_Guide.asciidoc

```diff
@@ -0,0 +1,381 @@
+
+
+RADIUS Certificate generation
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Using the PKI of Microsft include that all your certificates will be deliver by the root CA of the PKI. The public key of this certificate is necessary on PacketFence. 
+
+As for RADIUS authentication we will need to generate a certificate for PacketFence.
+
+To generate the RADIUS certificate the template `WebServer` will be used.
+
+The idea now is to create the request (.CSR), a private key from the PacketFence server and submit the CSR to the NDES server. Connect to PacketFence via ssh and type the following in the CLI to generate the CSR and sign it with the private key:
+
+ openssl req -new -newkey rsa:2048 -nodes -keyout server.key -out server.csr
+
+You will be prompt for some information, here is an example of a valid configuration.
+
+* CN=packetfence.local
+* C=CA
+* ST=QC
+* Locality=Montreal
+* Organization=Inverse
+* Organization Unit=IT
+
+No fields are mandatory else than the CN.
+
+Once you have your CSR we will submit it to the NDES server.
+
+Display the content of the .csr file and open a web browser, reach the following:
+
+ http://<ServerDNSName>/CertSrv/
+
+You can now submit a request. Click `Request a certificate`, select `advanced certificate request`. Paste the content of your CSR file and select the template `Web Server`. Click `Submit`. On this page select `Base 64 encoded` and click `Download certificate`. 
+
+This will give you the certificate(public key) for PacketFence, you have the private key from the openssl genrsa command and now we need to download the CA.
+
+To do so use the website where we were ealier; 
+
+ http://<ServerDNSName>/CertSrv/
+
+Click `Download a CA certificate, certificate chain or CRL`, select your CA certificate in the list, select `Base 64` as the encoding method and finaly click `Download CA certificate`.
+
+Copy those files from the location you downloaded them to PacketFence.
+
+Step 2: Configuring PacketFence
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Certificate storage on PacketFence
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+ 
+It is recommended to create a separate directory to separate EAP-TLS certificates from server certificates:
+
+ # mkdir /usr/local/pf/conf/ssl/tls_certs/
+
+RADIUS EAP-TLS authentication requires three files, the CA certificate, the server certificate and the private key.
+
+Copy those files in your newly created folder:
+
+* Privatekey of the RADIUS server (~/fqdn.server.key)
+* Certificate for RADIUS (obtained from the CSR submitted)
+* Certificate CA (downloaded from NDES website)
+
+Ensure that the files are readable by "pf": 
+
+ # chown pf:pf /usr/local/pf/conf/ssl/tls_certs/*
+ 
+RADIUS EAP-TLS and MSPKI
+^^^^^^^^^^^^^^^^^^^^^^^^
+
+Using the PKI generated certificates requires editing the radius EAP configuration file.
+
+Edit the /usr/local/pf/conf/radiusd/eap.conf file and replace the following lines with references to your new certificates in the 'tls' configuration block:
```

Comment:
*Edit `/usr/local/pf/conf/radiusd/eap.conf` and replace

Less words more backticks